Add support of unicode symbols

puzan opened the pull request:
`lxml` library works only with unicode strings. It is impossible to use base python `str` with cyrillic symbols inside `lxml`:

```
ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters
```

Current fix adds decoding value parameter if it is not unicode.

cmprescott commented:
Do you mind adding a unit test for situation you encountered? In the meantime I will try to resolve the error Ansible 2.0.0.2 gives with the module.

puzan commented:
Ok. I will add tests for this case during the week.

puzan commented:
@cmprescott Hi! I have rewritten unicode solution. I have added correct support for `add_children` and `set_children` functions. Also I have added unicode test.

I have added function `decode`. It decodes `str`, `str` in lists and `str` in dict values to unicode. It doesn't support int or other types (as only strings are expected as values).

cmprescott commented:
You the man!
